src/elements/content-sharing/__tests__/useContacts.test.js

```diff
@@ -95,23 +105,91 @@ describe('elements/content-sharing/hooks/useContacts', () => {
-            return expect(contacts).resolves.toEqual(MOCK_CONTACTS_API_RESPONSE);
+            return expect(contacts).resolves.toEqual([
+                ...MOCK_CONTACTS_API_RESPONSE.entries,
+                ...MOCK_GROUP_CONTACTS_API_RESPONSE.entries,
+            ]);
         });
+
+        test('should set the value of getContacts() to an empty array when no results are found', () => {
+            const EMPTY_GROUPS = { entries: [] };
+            const EMPTY_USERS = { entries: [] };
+            getGroupsInEnterprise = jest.fn().mockImplementation((itemID, getGroupsInEnterpriseSuccess) => {
+                return getGroupsInEnterpriseSuccess(EMPTY_GROUPS);
+            });
+            getUsersInEnterprise = jest.fn().mockImplementation((itemID, getUsersInEnterpriseSuccess) => {
+                return getUsersInEnterpriseSuccess(EMPTY_USERS);
+            });
+            mockAPI = createAPIMock({ getGroupsInEnterprise }, { getUsersInEnterprise });
+            let fakeComponent;
+
+            act(() => {
+                fakeComponent = mount(<FakeComponent api={mockAPI} transformResponse={transformResponseSpy} />);
+            });
+            fakeComponent.update();
+
+            const btn = fakeComponent.find('button');
+            expect(btn.prop('onClick')).toBeDefined();
+            const contacts = btn.invoke('onClick')(MOCK_FILTER);
+
+            expect(handleSuccess).toHaveBeenCalledWith(EMPTY_GROUPS);
+            expect(handleSuccess).toHaveBeenCalledWith(EMPTY_USERS);
+            expect(transformResponseSpy).not.toHaveBeenCalled();
+            return expect(contacts).resolves.toEqual([]);
+        });
+
+        /**
+         * A successful API call will always return an entries array. However, the Flow definitions
+         * for GroupCollection and UserCollection mark "entries" as optional, so we still need to test
+         * for the hypothetical case in which the entries array is undefined.
+         */
+        test.each`
+            groupsResponse                      | usersResponse                 | resolvedResponse                            | description
+            ${undefined}                        | ${undefined}                  | ${[]}                                       | ${'both responses are undefined'}
+            ${{}}                               | ${{}}                         | ${[]}                                       | ${'both responses are defined, but do not contain an entries array'}
+            ${undefined}                        | ${MOCK_CONTACTS_API_RESPONSE} | ${MOCK_CONTACTS_CONVERTED_RESPONSE}         | ${'users response is defined, and groups response is undefined'}
+            ${MOCK_GROUP_CONTACTS_API_RESPONSE} | ${undefined}                  | ${MOCK_GROUP_CONTACTS_API_RESPONSE.entries} | ${'groups response is defined, and users response is undefined'}
+        `(
+            'should set the value of getContacts() to an empty array when $description',
```

Comment:
should this text be updated? in the last 2 test cases, the resolvedResponse is not an empty array